pkg/ccl/backupccl/restore_planning.go

```diff
@@ -2183,19 +2211,21 @@ func planDatabaseModifiersForRestore(
 		)
 	}
 
-	var maxSeenID descpb.ID
-	for _, desc := range sqlDescs {
-		if desc.GetID() > maxSeenID {
-			maxSeenID = desc.GetID()
-		}
-	}
-
 	shouldRestoreDatabaseIDs := make(map[descpb.ID]struct{})
 	for _, db := range restoreDBs {
 		shouldRestoreDatabaseIDs[db.GetID()] = struct{}{}
 	}
 
 	var extraDescs []catalog.Descriptor
+	// Do one pass through the sql descs to map the database to its public schema.
```

Comment:
Where are we checking that we are only mapping dbID to public schema? sqlDescs can also contain UDS.